File: moxy-compiler/src/main/java/moxy/compiler/viewstate/MigrationMethod.java
package moxy.compiler.viewstate;

import javax.lang.model.element.ExecutableElement;
import javax.lang.model.element.TypeElement;

class MigrationMethod {

    TypeElement clazz;

    ExecutableElement method;

    MigrationMethod(final TypeElement clazz,
        final ExecutableElement method) {
        this.clazz = clazz;
        this.method = method;
    }
}

File: moxy-compiler/src/main/java/moxy/compiler/viewstate/MigrationMethod.kt
package moxy.compiler.viewstate

import javax.lang.model.element.ExecutableElement
import javax.lang.model.element.TypeElement

/**
 * Represents view interface method without strategy annotation.
 * One statement will be generated in `EmptyStrategyHelper` based on data from this class.
 */
data class MigrationMethod constructor(
    val viewInterface: TypeElement,
    val method: ExecutableElement
)
